end2end - #226 rems integration and others

jLebioda opened the pull request.

vildead reviewed (approved):
tested using POST request.

Code looks nice from what I can tell.

pinarpink commented on core/models/user.py:
```diff
@@ -186,3 +187,30 @@ def can_edit_contract(self, contract):
+            dataset=dataset,
+            access_notes=notes
+        )
+        new_access.save()
```
Before we return can we add here:

```
 new_logbook_entry = Share(  dataset=dataset,            granted_on=datetime.now(), share_notes='rems-application:'+application)
new_logbook_entry.save
```

jLebioda replied:
looks good!

pinarpink reviewed (approved):
Thanks a lot @jLebioda. I had one comment on creating a logbook entry. Otherwise looks good to me.

jLebioda commented on core/migrations/0019_auto_20210616_1407.py:
```diff
@@ -0,0 +1,25 @@
+        migrations.AlterField(
+            model_name='access',
+            name='defined_on_locations',
+            field=models.ManyToManyField(blank=True, help_text='The dataset locations on which access is defined.', null=True, related_name='accesses', to='core.DataLocation', verbose_name='Data Locations'),
```
Exactly! Just corrected that in cff2f55
Thanks for spotting :)

jLebioda commented:
@pinarpink @vildead just implemented both suggestions, let's wait for CI to confirm the tests

jLebioda commented:
Ready to be reviewed again; I've included some other minor improvements (also from the previous MR #240):

 * added a modal when publishing a dataset (this will also publish relevant project/data_declaration etc.)
 * added unpublish buttons to Dataset, Partner, Project, Cohort (they will display a message "Do you want to continue? YES/NO" + "this will not unpublish the entry from other systems")
 * added a button "See in django-admin" for administrators on Dataset, Partner, Project, Cohort pages
 * the elu_accession is a read-only field for Dataset, Partner, Project, Cohort
 * the is_published field is completely hidden (there's "Publish"/"Unpublish" buttons and "See in django-admin")
 * when using docker-compose, the version on about and index pages will display "develop", and not "undefined" anymore
 * added editing PartnerRoles (solves #235)
 * added `linebreaks` filter to Project's lay summary and Dataset description (i.e. line breaks will be translated to <br>`)
 * added `API_KEY` field to User model (so that there is no need for looking for `GLOBAL_API_KEY` anymore, maybe in the future we could even remove it completely)
 * added `oidc_id` to User model
 * added both API_KEY and oidc_id to User form in django admin
 * added search box to datasets and projects in django admin
 * dataset and project pages in django admin will display more columns (so that it's not just 1 column with `__str__()`, but title, acronym, elu_id etc.

vildead commented:
Great job @jLebioda! :1st_place_medal: 
I have added one humble commit :D 
I have tested the POST request and other features - works like a charm.
